fractions/microprofile/microprofile-jwt/src/main/java/org/wildfly/swarm/microprofile/jwtauth/runtime/MPJWTAuthExtensionArchivePreparer.java

```diff
@@ -174,192 +141,9 @@ public void process() throws Exception {
-                log.debugf("SecurityConstraint(%s), roles=%s, isPermitAll=%s", sc.urlPattern(), sc.roles(), sc.isPermitAll());
-            }
-        }
-        scannedClasses.add(classInfo.name());
-    }
-
-    private SecurityConstraint createSecurityConstraint(WebXmlAsset webXml, String uriPath, String[] localRoles) {
-        SecurityConstraint securityConstraint = webXml.protect(uriPath);
-        if (localRoles == null) {
-            // No roles == @DenyAll
-            securityConstraint.withRole("");
-        } else if (localRoles.length == 0) {
-            // Empty roles == @PermitAll
-            securityConstraint.permitAll();
-        } else {
-            securityConstraint.withRole(localRoles);
-        }
-        return securityConstraint;
-    }
-
-    private String getUriPath(String subpath, String fullAppPath) {
-        String uriPath;
-        if (subpath.isEmpty()) {
-            // Remove the trailing slash
-            uriPath = fullAppPath.substring(0, fullAppPath.length() - 1);
-        } else if (subpath.charAt(0) == '/') {
-            uriPath = fullAppPath + subpath.substring(1);
-        } else {
-            uriPath = fullAppPath + subpath;
-        }
-        // If this uri includes a path param, truncate and add a wildcard
-        int pathParamStart = uriPath.indexOf('{');
-        if (pathParamStart >= 0) {
-            uriPath = uriPath.substring(0, pathParamStart);
-            if (uriPath.charAt(uriPath.length() - 1) != '/') {
-                uriPath += '/';
-            }
-            uriPath += "*";
-        }
-        return uriPath;
-    }
-
-    private List<MethodInfo> getResourceMethods(ClassInfo classInfo, Iterable<DotName> httpMethods) {
-        List<MethodInfo> resourceMethods = new ArrayList<>();
-        for (MethodInfo method : classInfo.methods()) {
-            if (isResourceMethod(method, httpMethods)) {
-                resourceMethods.add(method);
-            }
-        }
-        return resourceMethods;
-    }
-
-    private boolean isResourceMethod(MethodInfo method, Iterable<DotName> httpMethods) {
-        for (DotName httpMethod : httpMethods) {
-            if (method.hasAnnotation(httpMethod)) {
-                return true;
-            }
-        }
-        return false;
-    }
 
-    private Iterable<DotName> collectHttpMethods() {
-        List<DotName> httpMethods = new ArrayList<>();
-        httpMethods.add(GET);
-        httpMethods.add(POST);
-        httpMethods.add(PUT);
-        httpMethods.add(DELETE);
-        httpMethods.add(HEAD);
-        httpMethods.add(OPTIONS);
-        for (AnnotationInstance customHttpMethod : index.getAnnotations(HTTP_METHOD)) {
-            httpMethods.add(customHttpMethod.name());
-        }
-        return httpMethods;
+        JAXRSArchive jaxrsArchive = archive.as(JAXRSArchive.class);
+        jaxrsArchive.findWebXmlAsset()
+                .setContextParam("resteasy.providers", MpJwtFilterRegistrator.class.getName());
```

Comment:
Is this required, if the class is annotated `@Provider`?

Reply:
It is not working for me without it, although the javadoc says it should: https://docs.oracle.com/javaee/7/api/javax/ws/rs/container/DynamicFeature.html

Pavol did the same for opentracing:
https://github.com/michalszynkiewicz/wildfly-swarm/blob/10ff026f7cec05bb938ac9c066b21191e9770a1e/fractions/microprofile/microprofile-opentracing/src/main/java/org/wildfly/swarm/mpopentracing/runtime/TracingInstaller.java#L50-L51

https://github.com/michalszynkiewicz/wildfly-swarm/blob/10ff026f7cec05bb938ac9c066b21191e9770a1e/fractions/microprofile/microprofile-opentracing/src/main/java/org/wildfly/swarm/mpopentracing/deployment/MicroprofileServerTracingFeature.java#L33


I think `@Provider` can be removed in this case, I'll test it and update the PR if it can.

Reply:
Ah I see. I think this is fine, was just curious!

Reply:
it worked without `@Provider` so I removed it.

Reply:
Ah maybe one other thing here, what happens if the `web.xml` from deployment already contains the `resteasy.providers` context param, with a user-provided value?

Reply:
I didn't think about it, I'll check it after PTO but I guess should merge it with what user provided.

Reply:
done